### PR TITLE
Change `ProxyCallback`s to return no value

### PR DESCRIPTION
**Pull request by jblebrun**
There's no need for a Store to act immediately on a result from a
callback to the StorageProxy (or other callers). In fact, there's
already no usage of this return value.

If a `StorageProxy` (or other Store agent) encounters an issue applying
the callback message from a `Store`, it doesn't need to signal that
immediately, instead it can send the appropriate message back to the
`Store` (as is already happening).

This also removes `ParcelableProxyMessageChannel` (rather than also
refactoring it) since it was not being used anymore, except for in a few
tests.

### CODE REVIEW

File: java/arcs/android/storage/service/IStorageServiceCallback.aidl [hunk 1]
```diff
@@ -17,5 +17,5 @@ import arcs.android.storage.service.IResultCallback;
 /** Variant of ProxyCallback intended for StorageService communications. */
 interface IStorageServiceCallback {
     /** Handles an incoming ProxyMessage. */
-    void onProxyMessage(in ParcelableProxyMessage message, IResultCallback resultCallback);
+    void onProxyMessage(in ParcelableProxyMessage message);
```
Comment (cromwellian) [inline java/arcs/android/storage/service/IStorageServiceCallback.aidl]: Shouldn't you add the 'oneway' keyword?
"The oneway keyword modifies the behavior of remote calls. When used, a remote call does not block; it simply sends the transaction data and immediately returns. The implementation of the interface eventually receives this as a regular call from the Binder thread pool as a normal remote call. If oneway is used with a local call, there is no impact and the call is still synchronous."

Comment (jblebrun) [reply java/arcs/android/storage/service/IStorageServiceCallback.aidl]: Yes you're right, I think that makes sense here. 

Do you think it makes sense to apply `oneway` to the other methods in the interface which are `void` and provide a callback for getting a response?

Comment (cromwellian) [reply java/arcs/android/storage/service/IStorageServiceCallback.aidl]: I don't know the underlying Android implications (ian may know), but anything that tells AIDL/Binder "don't bother waiting, fire and forget" seems like a good idea to me.

File: java/arcs/android/storage/service/DeferredProxyCallback.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package arcs.android.storage.service
+
+import arcs.android.storage.ParcelableProxyMessage
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.Deferred
+
+/**
+ * Provides an object that can be provided to service calls as an [IStorageService.Stub],
+ * and then waiting on as a deferred. Useful when you want a single message response.
```
Comment (csilvestrini) [inline java/arcs/android/storage/service/DeferredProxyCallback.kt]: `waiting on` -> `waited on`

File: javatests/arcs/core/storage/RamDiskBackingStoreIntegrationTest.kt [hunk 3]
```diff
@@ -68,7 +68,7 @@ class RamDiskBackingStoreIntegrationTest {
         val count2 = CrdtCount()
         count2.applyOperation(MultiIncrement("them", version = 0 to 10, delta = 15))
 
-        val id = store.on(MultiplexedProxyCallback { _, _ -> true })
+        val id = store.on(MultiplexedProxyCallback { _, _ -> })
```
Comment (csilvestrini) [inline javatests/arcs/core/storage/RamDiskBackingStoreIntegrationTest.kt]: No need for the args and arrow, you can just use empty braces `{ }`

Comment (jblebrun) [reply javatests/arcs/core/storage/RamDiskBackingStoreIntegrationTest.kt]: I had tried that; passing `{}` to `MultiplexedProxyCallback` here results in a compiler error for me. It seems that parameter omission is only supported when there is one argument.

I can, however, simply omit this `store.on` call completely.

**Pull request merged**